Mono.Debugging/Mono.Debugging.Client/DebuggerSession.cs

```diff
@@ -1300,11 +1309,15 @@ internal protected void OnTargetDebug (int level, string category, string messag
 			}
 		}
 
-		internal protected void OnAssemblyLoaded (string assemblyLocation)
+		internal protected void OnAssemblyLoaded (Assembly assembly)
 		{
-			AssemblyLoaded?.Invoke (this, new AssemblyEventArgs (assemblyLocation));
+			lock (assemblies) {
+				assemblies.Add (assembly);
+			}
+
+			AssemblyLoaded?.Invoke (this, new AssemblyEventArgs (assembly.Path));
```

Comment:
It would be better to add a reference to the Assembly instance to AssemblyEventArgs, instead of just the assembly path.

Reply:
The existing event doesn't need to change, a new property and constructor can be added to AssemblyEventArgs